### PR TITLE
Add post position integration test

### PR DESCRIPTION
**Pull request by jimmordino**
testing post position sproc, nullify position on delete

@jogura @armonr

### CODE REVIEW

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java [hunk 1]
```diff
@@ -405,6 +405,53 @@ public void undeletePost() throws Exception { //NOSONAR
   }
 
   @Test
+  public void postPositions() throws Exception { //NOSONAR
+    final Long TEST_THREAD_ID = 1840415539986433081L;
```
Comment (WikiaTech) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00117)

File: service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java [hunk 2]
```diff
@@ -405,6 +405,53 @@ public void undeletePost() throws Exception { //NOSONAR
...
+    final Integer TEST_SITE_ID = SITE_ID_GLEE;
```
Comment (WikiaTech) [inline service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00117)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues:
* MAJOR 3 major
* MINOR 16 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L70) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L72) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L87) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L82) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L66) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L78) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L76) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L68) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MAJOR [Define and throw a dedicated exception instead of using a generic one.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L94) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/8b922e553f846d0657450c7ecce1debdb8ddfd01/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L497) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)

**Comment by jogura** [comment]
:+1:

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues:
* MINOR 16 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L71) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L82) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L76) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L69) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L501) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/3d20dd8ca69094b6e99bf05232fc03b461f01e35/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

**Comment by armonr** [comment]
LGTM

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues:
* MINOR 16 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L77) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L83) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L74) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L80) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* INFO [Complete the task associated to this TODO comment.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L501) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L73) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L71) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L75) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L76) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L81) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L79) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L84) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L67) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L85) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L88) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L69) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)
* MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/Wikia/pandora/blob/8ca6c43e6404e0b5b5990332447e5e4f70cdf652/service/discussion/src/test/java/com/wikia/discussionservice/resources/PostResourceIntegrationTest.java#L82) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1213)

**Pull request merged**